server/services/ai.ts

```diff
@@ -36,6 +188,780 @@ export default class AIService {
         player.credits = Math.max(0, player.credits);
     }
 
+    async _doAdvancedLogic(game: Game, player: Player, isFirstTick: boolean, isLastTick: boolean) {
+        // Considering the growing complexity of AI logic, 
+        // it's better to catch any possible errors and have the game continue with disfunctional AI than to break the game tick logic.
+        try {
+            let aiState: AiState;
+            if (isFirstTick || !player.ai || !player.aiState) {
+                aiState = this._setupAi(game, player);
```

Comment:
We should perform this initial setup in (or called by) the `play` function separately as it could apply to both basic and advanced AI.

Reply:
Hmm. `_setupAi` only creates the AI state object, which is exclusively for the advanced AI. I've renamed the function now though to reflect that better.